### ISSUE TITLE
Modifier la méthode de création des attributs géométriques lors de la création des tables

### ISSUE THREAD

**Issue by MaelREBOUX** [3liz/QgisCadastrePlugin#241]
Pointé par @mdouchin  dans https://github.com/3liz/QgisCadastrePlugin/issues/237#issuecomment-669824723

> Nous devons en effet adapter le code et ne plus utiliser cette fonction (issue d'un historique), mais directement créer les géométries via la syntaxe `geometry(Point, 2154)`

Tous les attributs géométriques créés par les requêtes SQL disponibles dans https://github.com/3liz/QgisCadastrePlugin/tree/master/scripts/plugin sont en effet créés après la création de la table, en 2 temps.

Exemple :

```sql
CREATE TABLE geo_commune
(
  geo_commune text NOT NULL,
  annee text NOT NULL,
  object_rid text,
  idu text,
  tex2 text,
  creat_date date,
  update_dat date,
  commune text,
  lot text,
  ogc_fid serial NOT NULL
);
SELECT AddGeometryColumn ( current_schema::text, 'geo_commune', 'geom', 2154 , 'MULTIPOLYGON', 2 );
```

**Comment by MaelREBOUX** [comment]
Question pour @mdouchin à propos des systèmes de coordonnées.

Dans commun_create_metier.sql toutes les tables sont créées en Lambert93 (2154) (exemple avec [geo_commune](https://github.com/3liz/QgisCadastrePlugin/blob/master/scripts/plugin/commun_create_metier.sql#L830-L843))

Puis on charge les données EDIGEO via la fonction [importEdigeoThfToDatabase](https://github.com/3liz/QgisCadastrePlugin/blob/master/cadastre_import.py#L1292) dans cadastre_import.py
A cette étape on récupère et utilise 2 variables sourceSridFull et targetSridFull récupérées depuis l'interface QGIS.

Vu le code j'en déduis qu'on charge (pour notre cas) des données en 3948 alors que les tables sont spécifiées en 2154.
Or, quand je regarde après le chargement, l'attribut `geom` est bien en 3948 pour toutes les tables / couches.

A quel moment la redéfinition du système de projection se fait ?

**Comment by Gustry** [comment]
Je suppose qu'il y a un remplacement à la volée du système de projection dans le fichier SQL quand il est lu par Python : 
https://github.com/3liz/QgisCadastrePlugin/blob/89b8e11247d0b08ae34cd1fc09e19cc871aa54d7/cadastre_import.py#L244
et 10 lignes plus bas
https://github.com/3liz/QgisCadastrePlugin/blob/89b8e11247d0b08ae34cd1fc09e19cc871aa54d7/cadastre_import.py#L254

**Comment by MaelREBOUX** [comment]
okay ! merci @Gustry 

Pour savoir si on devait retoucher la gestion des projections en passant ou pas... Donc : pas.
Mais on va rajouter une ligne de commentaire ;)

@EtienneRouvin est sur le coup

**Comment by MaelREBOUX** [comment]
@mdouchin ou @Gustry  Vous pourriez nous redire où se trouve le fichier / la fonction qui reformate les requêtes SQL natives PostGIS en requêtes compatibles spatialite SVP ?

**Comment by landryb** [comment]
@EtienneRouvin quel est le fix potentiel finalement ? Je vais bientot devoir importer mes 12 départements dans postgis, et je passe sur la v3 justement... donc si je peux tester un fix je suis preneur

**Comment by landryb** [comment]
si je comprends bien `AddGeometryColumn()` existe toujours en postgis 3 (cf https://postgis.net/docs/AddGeometryColumn.html) c'est juste la signature qui est mauvaise dans les scripts SQL, et il faut rajouter des casts pour être compatible avec postgis 2 / postgis 3 / spatialite ?

**Comment by MathRobin** [comment]
T'as essayé avec une base postgre aws rds ? T'as quoi comme versions exactes ? Postgre, plugins, derniers fichiers de cadastre dispo ? Mon but dans #237 c'est de pouvoir importer les données dans une aws rds

**Comment by landryb** [comment]
non, c'est du postgres/postgis venant des paquets debian pgdg, jamais joué avec aws et encore moins rds. le ticket #226 montre qu'il y'a aussi ce pbm sur du postgis 2.1, donc je pense que ce n'est pas forcement lié a la version de postgis.. mais aux données ? A la version de python utilisée qui ne traite pas forcement les types de la meme facon, et requiert des casts dans certains cas ? mon test était sur une seule commune du dept 38, je vais voir en important un département..

**Comment by MathRobin** [comment]
Teste avec le département 33 stp, c'est dessus que je galère

**Comment by landryb** [comment]
oui ben désolé, je trouve que c'est déja pas mal d'avoir 12 départements, et le 33 n'en fait pas partie ;)

**Comment by MathRobin** [comment]
12 sont passés sans soucis ? Mais c'est trop injuste XD je vais réessayer avec un fichier de cadastre plus récent pour voir

**Comment by landryb** [comment]
euh, il faut un 'certain temps' pour importer 12 départements, mais jusqu'ici il m'en a fait 7 sans soucis manifeste.. de toute facon, une fois que la table est créee correctement, je pense que tout le reste roule tout seul;

**Comment by MathRobin** [comment]
Ça marche, je vais essayer avec un autre. Lesquels sont passés chez toi déjà ? Que je tente avec un de ceux là ?

**Comment by landryb** [comment]
01 03 07 15 26 38 42 43 depuis hier 15h, actuellement 63, reste 69 73 74..

**Comment by MathRobin** [comment]
OK merci pour les infos !

**Comment by MathRobin** [comment]
A l'époque j'avais regardé la Gironde (où je vis, j'ai fait simple). Par contre la version de postgis, aucune idée...
